src/Datasource/ResultSetDecorator.php

```diff
@@ -46,4 +46,14 @@ public function count(): int
 
         return count($this->toArray());
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function __debugInfo(): array
+    {
+        $parentInfo = parent::__debugInfo();
+
+        return array_merge($parentInfo, ['items' => $this->toArray()]);
```

Comment:
My only concern here is that dumping a large set could get expensive memory wise. Limiting the number of results would reduce the utility of the debug output though.

Reply:
@dereuromark whats you opinion on this?
We already limit the output of the variables panel by not showing all nested properties so I wouldn't mind also limiting the amount of items being printed. But I would also make it configurable like `DebugKit.variablesPanelMaxDepth`